inventory/main/src/main/java/org/opennms/horizon/inventory/service/MonitoringSystemService.java

```diff
@@ -47,25 +45,40 @@ public Optional<MonitoringSystemDTO> findBySystemId(String systemId, String tena
         return systemRepository.findBySystemIdAndTenantId(systemId, tenantId).map(mapper::modelToDTO);
     }
 
-    public void addMonitoringSystemFromHeartbeat(TenantLocationSpecificHeartbeatMessage message) throws LocationNotFoundException {
+    public Optional<MonitoringSystemDTO> findById(long systemId, String tenantId) {
+        return systemRepository.findByIdAndTenantId(systemId, tenantId).map(mapper::modelToDTO);
+    }
```

Comment:
Does this method search by `id` or `systemId`? They look to be two distinct values in the table. I assume that the first parameter should be `id` given its type.

```suggestion
    public Optional<MonitoringSystemDTO> findById(long id, String tenantId) {
        return systemRepository.findByIdAndTenantId(id, tenantId).map(mapper::modelToDTO);
    }
```